test/tests/ref.test.ts

```diff
@@ -0,0 +1,146 @@
+    expect((RefTest.schema.path('refFieldString2') as any).options.ref).to.equal('RefTestString');
+
+    expect((RefTest.schema.path('refArrayString') as any).instance).to.equal('Array');
+    expect((RefTest.schema.path('refArrayString') as any).caster.instance).to.equal('String');
+    expect((RefTest.schema.path('refArrayString') as any).caster.options.ref).to.equal('RefTestString');
+    expect((RefTest.schema.path('refArrayString2') as any).instance).to.equal('Array');
+    expect((RefTest.schema.path('refArrayString2') as any).caster.instance).to.equal('String');
+    expect((RefTest.schema.path('refArrayString2') as any).caster.options.ref).to.equal('RefTestString');
+  });
+
+  it('check generated ref schema for number _id', async () => {
+    expect((RefTest.schema.path('refFieldNumber') as any).instance).to.equal('Number');
+    expect((RefTest.schema.path('refFieldNumber') as any).options.ref).to.equal('RefTestNumber');
+    expect((RefTest.schema.path('refFieldNumber2') as any).instance).to.equal('Number');
+    expect((RefTest.schema.path('refFieldNumber2') as any).options.ref).to.equal('RefTestNumber');
+
+    expect((RefTest.schema.path('refArrayNumber') as any).instance).to.equal('Array');
+    expect((RefTest.schema.path('refArrayNumber') as any).caster.instance).to.equal('Number');
+    expect((RefTest.schema.path('refArrayNumber') as any).caster.options.ref).to.equal('RefTestNumber');
+    expect((RefTest.schema.path('refArrayNumber2') as any).instance).to.equal('Array');
+    expect((RefTest.schema.path('refArrayNumber2') as any).caster.instance).to.equal('Number');
+    expect((RefTest.schema.path('refArrayNumber2') as any).caster.options.ref).to.equal('RefTestNumber');
+  });
+
+  it('check generated ref schema for Buffer _id', async () => {
+    expect((RefTest.schema.path('refFieldBuffer') as any).instance).to.equal('Buffer');
+    expect((RefTest.schema.path('refFieldBuffer') as any).options.ref).to.equal('RefTestBuffer');
+    expect((RefTest.schema.path('refFieldBuffer2') as any).instance).to.equal('Buffer');
+    expect((RefTest.schema.path('refFieldBuffer2') as any).options.ref).to.equal('RefTestBuffer');
+
+    expect((RefTest.schema.path('refArrayBuffer') as any).instance).to.equal('Array');
+    expect((RefTest.schema.path('refArrayBuffer') as any).caster.instance).to.equal('Buffer');
+    expect((RefTest.schema.path('refArrayBuffer') as any).caster.options.ref).to.equal('RefTestBuffer');
+    expect((RefTest.schema.path('refArrayBuffer2') as any).instance).to.equal('Array');
+    expect((RefTest.schema.path('refArrayBuffer2') as any).caster.instance).to.equal('Buffer');
+    expect((RefTest.schema.path('refArrayBuffer2') as any).caster.options.ref).to.equal('RefTestBuffer');
+  });
+
+  it('check reference with string _id', async () => {
+    const id1: string = 'testid1';
+    const id2: string = 'testid2';
+
+    const t = new RefTest();
+    t.refFieldString = id1; // should not throw compilation error
```

Comment:
could you remove `// should not throw compilation error` on all lines?

Reply:
Yes. The idea was that these lines are actually test of the Ref declaration being correct. Otherwise test compilation would fail on these lines